layer.md

```diff
@@ -311,11 +311,10 @@ Any given image is likely to be composed of several of these Image Filesystem Ch
 Certain layers, due to legal requirements, may not be regularly distributable.
 Typically, such layers are downloaded directly from a distributor but are never uploaded.
 
-Layers that have these restrictions SHOULD be tagged with an alternative mediatype of `application/vnd.oci.image.layer.nondistributable.v1.tar+gzip`.
+Layers that have these restrictions SHOULD be tagged with an alternative media type of `application/vnd.oci.image.layer.nondistributable.v1.tar`.
```

Comment:
This is true of many instances where I currently only talk about the uncompressed type (which is one reason why I prefer the structured syntax suffix approach from #332).  With a structured syntax suffix off the table, this PR tries to stay clean by only defining the `+gzip` forms in `media-types.md`.  For example, this section defines a nondistributable, uncompressed layer.  If you're wondering what a compressed, nondistributable layer is, you look it up [in the media type list][1], and see that it's a gzipped version of the uncompressed, nondistributable layer (linking to this section).  You can also look up the uncompressed, nondistributable layer in the media-type table, and that also links to this section.

If, on the other hand, you're reading this section and think “hmm, I wonder what media type this would be if I gzip it?”, the current state of this PR requires to you to grep or remember that we define `+gzip` forms for these specific types.  If we want to continually remind readers that we define those `+gzip` types, I'll reluctantly inject whatever boilerplate you need in most locations where we reference an uncompressed layer type; just tell me what you want that boilerplate to be.  Maybe `(or ['…+gzip'][media-types.md#oci-image-media-types)` almost everywhere we list an uncompressed-layer media type?  Do you really want that?

[1]: https://github.com/wking/image-spec/blob/fd9719867cf82ebac6d26bd9212e8bb92a9fd20b/media-types.md#oci-image-media-types